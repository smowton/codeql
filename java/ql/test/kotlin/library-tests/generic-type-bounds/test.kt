class Test<T : Number, S : T, R> {

  fun <Q : Number> f() { }

}

class MultipleBounds<P> where P : List<String>, P : Set<String> { }
