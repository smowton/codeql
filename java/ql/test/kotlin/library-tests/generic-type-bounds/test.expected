classTVs
| test.kt:1:12:1:21 | T | test.kt:1:1:5:1 | Test | Number |
| test.kt:1:24:1:28 | S | test.kt:1:1:5:1 | Test | T |
| test.kt:1:31:1:31 | R | test.kt:1:1:5:1 | Test | Object |
| test.kt:7:22:7:22 | P | test.kt:7:1:7:67 | MultipleBounds | List<String> |
| test.kt:7:22:7:22 | P | test.kt:7:1:7:67 | MultipleBounds | Set<String> |
functionTVs
| test.kt:3:8:3:17 | Q | test.kt:3:3:3:26 | f | Number |
